pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -151,15 +152,28 @@ type CephHealthMessage struct {
 	Message  string `json:"message"`
 }
 
-type ClusterState string
+type RookConditions struct {
+	Type               ConditionType      `json:"type,omitempty"`
+	Status             v1.ConditionStatus `json:"status,omitempty"`
+	Reason             string             `json:"reason,omitempty"`
+	Message            string             `json:"message,omitempty"`
+	LastHeartbeatTime  metav1.Time        `json:"lastHeartbeatTime,omitempty"`
+	LastTransitionTime metav1.Time        `json:"lastTransitionTime,omitempty"`
+}
+
+type ConditionType string
 
 const (
-	ClusterStateCreating   ClusterState = "Creating"
-	ClusterStateCreated    ClusterState = "Created"
-	ClusterStateUpdating   ClusterState = "Updating"
-	ClusterStateConnecting ClusterState = "Connecting"
-	ClusterStateConnected  ClusterState = "Connected"
-	ClusterStateError      ClusterState = "Error"
+	ConditionIgnored     ConditionType = "Ignored"
+	ConditionConnecting  ConditionType = "Connecting"
+	ConditionConnected   ConditionType = "Connected"
+	ConditionProgressing ConditionType = "Progressing"
+	ConditionReady       ConditionType = "Ready"
+	ConditionNotReady    ConditionType = "Not Ready"
+	ConditionAvailable   ConditionType = "Available"
+	ConditionFailure     ConditionType = "Failure"
+	ConditionExpanding   ConditionType = "Expanding"
+	ConditionUpgrading   ConditionType = "Upgrading"
```

Comment:
Can we add `Shrinking`?

Reply:
Does ceph support shrinking of a cluster? I looked up and didn't see any reference to it.

Reply:
Okay. I'll add the Shrinking also.

Reply:
Expanding and Shrinking I removed it for now. Just included the basic conditions for now.

Reply:
Why?

Reply:
Because everytime a cluster is updating or upgrading the osd pods also starts its reconcilation. This may also be detected as expansion. So I need to figure out a logic to detect this. I have something I am thinking may work. 

But in a meeting @travisn said lets get in these basic conditions and I can work on building on top of that. That's why I pushed this in. @leseb

Reply:
> Some of the nits could have been avoided by a linter... Make sure to use one on your IDE.

Yes.